docs/pageserver-tenant-migration.md

```diff
@@ -0,0 +1,20 @@
+## Pageserver tenant migration
+
+### Overview
+
+This feature allows to migrate a timeline from one pageserver to another by utilizing remote storage capability.
+
+### Migration process
+
+Pageserver implements two new http handlers: timeline attach and timeline detach.
+Timeline migration is performed in a following way:
+1. Timeline attach is called on a target pageserver.
+2. For now it is necessary to manually initialize replication stream via callmemaybe call so target pageserver initializes replication from safekeeper (it is desired to avoid this and initialize replication directly in attach handler, but this requires some refactoring (probably [#997](https://github.com/zenithdb/zenith/issues/997)/[#1049](https://github.com/zenithdb/zenith/issues/1049))
+3. Replication state can be tracked via timeline detail pageserver call.
+4. Compute node should be restarted with new pageserver connection string.
```

Comment:
Is it possible that old and new compute nodes exist simultaneously?
Which component is responsible for choosing the primary node - the one that serves write queries and streams WAL to safekeepers?
It's worth describing it in README.

Reply:
It is solved on the safekeepers level. Two computes cannot simultaneously exist. If we support config change on the fly there shouldn't be even a pod restart. Also in the future it is planned to support this change without compute restart whatsoever. In its current state responsibility for rescheduling the compute with updated config is on console.

Thanks for pointing that out, I'll add this to the doc

Reply:
Added to the doc, feel free to unresolve if something is still missing